### DISCUSSION TITLE
Introductory Tutorial

### DISCUSSION THREAD

**Discussion by gramian** [ArcadeData/arcadedb#1654] [Ideas]
What should be in an ArcadeDB introductory tutorial?

Please comment.

Thanks!

**Comment by ygyg70** [comment]
When using the engine in embedded mode with Gremlin Java, which Maven dependencies are needed, what is the correct way to create a database, how to configure, how to use it, use Gremlin, handle transactions etc.

**Comment by eisenhaus335** [comment]
A common newbie tutorial that could be developed into advanced idea. For example, ledger, event sourcing, CQRS

ArcadeDB designed as multi-model database but their fundamental mechanic are Edges. This is very different than common database tutorial with relationship key and 'tabular'. So, how does we reorienting into arcadedb design, that is different than other graph but also not like tabular database. Also, how we use common ArcadeDB operator (for example aggregator, list, string manipulator) to compose interpretable data while also 'make sense' to maintain them in relation to how Arcade multi-model database

**Comment by kumsumit** [comment]
This is the code for running Embedded server
```java
ContextConfiguration config = new ContextConfiguration();
ArcadeDBServer server = new ArcadeDBServer(config);
server.start();
```
`server.start()` gives complaining about  `no symbol start found`
Please improve the documentation for Embedded server
Also add, where does database file  gets stored when it is being created by the browser (i.e. from this url `http://127.0.0.1:2480/#`)  .
I have created database from the browser but trying to find in the database folder I found it as empty but the folder is empty but I am able to see it through my browser and able to work using my application. I have to move that file to server for the production.
But I am unable to do, because of lack of knowing where it is stored

**Reply by gramian** [reply]
Hi,
I cannot say what is wrong with your code, but maybe it helps to compare with a test also using the server, for example: https://github.com/ArcadeData/arcadedb/blob/main/server/src/test/java/com/arcadedb/server/SelectOrderTest.java#L74 . Is maybe the import `import com.arcadedb.server.ArcadeDBServer;` missing?

The databases should be under the `databases` folder below the server base folder by default. When I create a new database in studio, it immediately appears there. Please file an issue if this problem persists.

**Reply by kumsumit** [reply]
My mistake, pardon me. I was doing it in outside of the class, I was using earlier Remote Connection , while changing the code I have just copy pasted from the documentation, at that time I just forget that calling method should be in method. Realized, after going through the code shared by you.  Thanks for realizing me my trivial mistake , I also found the database file later after posting here , it was on my home folder, finding after taking backup of the file and then searching it my whole system.

**Comment by JusticeN** [comment]
Springboot and quarkus are One of the most used Java Library today.

It will be Great to show how to integrate with these libraries.

With common associated topics like security, replication, database transaction, datastore/orm(hibernate, jpa, spring data, ...), metrics, versioning, ...

Show how to use multimodal db in real world. For example an app with database, caching and metrics which use Postgres, redis and Prometheus. Then replace all the 3 by just adding Arcadedb dependency.

How performance improves 

And show how Arcadedb can replace all that in an embedded single spring boot using other existing libraries.

I think after an example like that Arcadedb will be qualified for the semifinals of software architecture 🎉

Arcadedb is a great ideas I can not just use it for production now because we are not sure if it will really match the expectations. 

"One db to replace them all, using existing libraries. One jar file with everything included. From database, cache to timeseries"
